### PR TITLE
Extend Contract class: call arbitrary contract getters

### PR DESCRIPTION
**Pull request by tallywiesenberg**
I've expanded the Contract class to be able to call arbitrary contract getter functions. In other words, it can call contract read functions w/o being specific to an ABI. Contract.read() accepts the name of the getter (str) and the arguments of the getter (**kwargs). Under the hood, read() searches the ABI of the connected contract for that function by name, builds a web3 callable function (if that function is found), then reads from the contract using the getter.

Plans moving forward:
1. develop Contract.write() method
2. standardize build_tx() and submit_tx() in contract class (internal functions?) 
3. wrap our common TellorX function calls into a TellorXWrapper child class of Contract

Can be integrated into:
- tellormon
- reporter

### CODE REVIEW

File: src/telliot/utils/contract.py [hunk 1]
```diff
@@ -23,17 +24,100 @@ class Contract(Base):
...
+
+        returns:
+        Tuple: contains contract function outputs (tuple) and success (bool)
+        """
+
+        if self.contract:
+            try:
+                contract_function = self.contract.get_function_by_name(func_name)
+                print(contract_function)
+                return (contract_function(**kwargs).transact(), True)
```
Comment (clamdad) [inline src/telliot/utils/contract.py:61]: remove redundant parentheses (on all tuple returns in this file)

File: src/telliot/utils/contract.py [hunk 2]
```diff
@@ -23,17 +24,100 @@ class Contract(Base):
...
+    #         )
+
+    #         _ = self.node.web3.eth.wait_for_transaction_receipt(tx_hash, timeout=360)
+
+    #         return True
+    #     except Exception:
+    #         print("tx was unsuccessful")
+    #         return False
+
+    def listen() -> None:
```
Comment (clamdad) [inline src/telliot/utils/contract.py:121]: method must have `self` as first input

File: src/telliot/utils/contract.py (hunk 1 above)
Comment (tallywiesenberg) [reply src/telliot/utils/contract.py:61]: Thank you!

**Comment by tallywiesenberg** [comment]
I need to add assertions to `test_call_read_function()` and implement @clamdad 's fixes. I will re-request after that!

**Review by clamdad — APPROVED** [review]

**Pull request merged**